server/src/main/java/fr/uga/l3miage/tp4/repositories/CandidateRepository.java

```diff
@@ -0,0 +1,30 @@
+package fr.uga.l3miage.tp4.repositories;
+import fr.uga.l3miage.tp4.enums.TestCenterCode;
+import fr.uga.l3miage.tp4.models.CandidateEntity;
+import fr.uga.l3miage.tp4.models.ExamEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+
+@Repository
+public interface CandidateRepository extends JpaRepository<CandidateEntity, Long> {
+
+    //Récupérer la liste des candidats du TestCenter ayant le code 'GRE'
+    Set<CandidateEntity> findByTestCenterEntityCode(TestCenterCode code);
```

Comment:
attention à la sémantique, ici tu les cherche tous, donc `findAll`
```suggestion
    Set<CandidateEntity> findAllByTestCenterEntityCode(TestCenterCode code);
```